### PR TITLE
Transaction signer selection

### LINKED ISSUE #405: Signer Panel (transactions)
When user is editing a Transaction the user needs to have a correct number of signatories selected.  

**Happy Path 👍**

🔹Single Signer Experience

One signer will be always selected by default - think radio button style 🔘

As a rule, the first account `0x01` will be the account selected "by default".   If a user selects another account they can do so, and the previous selection will be unselected, that way only **one account can be selected at a time** (see below):

image

Also the top message will change from `Default` to `1 Signer Selected`:
image

🔹Multiple Signer Experience

For multiple signers per transaction, the user will be guided to select additional signers - think a checkbox style ☑️

If less than the required amount of signers are selected, the Send button will be greyed out, and an orange circle with appear on top with the current amount selected:

image

**Assumptions**
- The playground backend knows exactly how many signers are required for each transaction to be executed. 
- The system remembers the signer choice for each transaction

### Questions:

**What if user has too many signers selected?**

Two cases:

**_1. A single signer is required_**

One signer will be always selected by default. The user can only choose one signer at a time. There is no way to unselect all signers.

**_2. More than one signer is required_**
The user is able to choose only the proper amount of signers. The signer has to be unselected to choose another signer.

**What if user has less than needed signers selected?**

The Send / Deploy button will be greyed out: 
image

What errors to display and where?

For now, to keep it simple use the number of errors inside the signer panel:
image

Notes:
  ### Transactions:
  - show header message when more signers are needed.
    - `2 of 3 signers` 
  - remove error in this case

### Design

for any additional questions refer to: https://www.figma.com/file/51JJzKlDh2O1nhZNuwzm3v/Playground-Hi-Fi%E2%80%99s-(Internal)?node-id=6%3A139678

signer panel states
https://www.figma.com/file/G7Z28scDlkNFvgDyR2nSnq/Playground-Library-%7C-Components-(External)?node-id=298%3A117977

User editing transaction
https://www.figma.com/file/G7Z28scDlkNFvgDyR2nSnq/Playground-Library-%7C-Components-(External)?node-id=298%3A125484

### PR DESCRIPTION
**Pull request by bthaile**
closes #405 of signer selection

Another ticket to address the button styling and error handling.

Pulled existing Signer component into new SingerPanel to add new functionality will remove old <Signer> component after new one is complete done and verified working correctly.

### CODE REVIEW

File: src/components/AccountAvatars.tsx [hunk 1]
```diff
@@ -2,27 +2,30 @@ import Avatar from 'components/Avatar';
...
-        mx="0.5rem"
+        mx="0.25rem"
         sx={{
-          position: 'relative',
-          borderRadius: '50%',
-          '.avatar': {
-            position: 'relative',
-            top: '-3px',
-          },
+          borderRadius: '0.5rem',
```
Comment (MrDSGC) [inline src/components/AccountAvatars.tsx]: Currently, the other components have a styles object saved like so: 
```
const styles = {
  root:{
    ...
  },
  thing: {
    ...
  } 
}
```

I see here you're plugging the object directly into the component. do you have any thoughts on the differences between using either methods

File: src/components/AccountAvatars.tsx [hunk 2]
```diff
@@ -37,14 +40,16 @@ export const AvatarList = ({ children }: ChildProps) => {
...
+        overflowX: 'scroll'
       }}
     >
       {children}
     </Flex>
   );
 };
 
-const noop = (): void => {};
+const noop = (): void => { };
```
Comment (MrDSGC) [inline src/components/AccountAvatars.tsx:59]: reasoning for this whitespace?

File: src/components/Arguments/index.tsx [hunk 3]
```diff
@@ -120,10 +121,10 @@ const getLabel = (
   return resultType === ResultType.Contract
     ? 'Deployment'
     : resultType === ResultType.Script
-    ? project.scriptTemplates[index].title
-    : resultType === ResultType.Transaction
-    ? project.transactionTemplates[index].title
-    : 'Interaction';
+      ? project.scriptTemplates[index].title
```
Comment (MrDSGC) [inline src/components/Arguments/index.tsx:124]: should this be a switch statement?

File: src/components/Icons/CollapseOpenIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+function CollapseOpenIcon() {
```
Comment (MrDSGC) [inline src/components/Icons/CollapseOpenIcon.tsx:3]: you can use `transform` to change the orientation of an icon. https://github.com/onflow/flow-playground/blob/mvp/src/components/Editor/FileExplorer/index.tsx#L50

File: src/components/SignersPanel/index.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      {indexes.map(index => {
+        return AvatarIcon(seed, index, complete);
+      })}
+    </Flex>
+  )
+}
+
+const PanelHeader = (maxSelection: number, accounts: Account[], seed: number, selected: number[] = []) => {
+  // TODO: accounts might be needed for displaying account addr, to be determined
+  let message = ""
```
Comment (MrDSGC) [inline src/components/SignersPanel/index.tsx:52]: ;

File: src/components/SignersPanel/index.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  const correctNumSigners = selected.length === maxSelection;
+  const signersSelected = "Signers Selected";
```
Comment (MrDSGC) [inline src/components/SignersPanel/index.tsx]: is it best practice to save text constant like this? and centralize constants in their own file?

 `const SIGNERS_SELECTED = "Signers Selected";`

File: src/components/Icons/CollapseOpenIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    return (
+        <svg width="10" height="6" viewBox="0 0 10 6" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <path d="M9 5L5 1L1 5" stroke="#2F353F" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" />
+        </svg>
+    )
+}
+
+export default CollapseOpenIcon;
```
Comment (MrDSGC) [inline src/components/Icons/CollapseOpenIcon.tsx:11]: missing line

File: src/components/SignersPanel/CollapseButton.tsx [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import CollapseClosedIcon from "components/Icons/CollapseClosedIcon";
+import CollapseOpenIcon from "components/Icons/CollapseOpenIcon";
+import React from "react";
+
+export const CollapseButton = ({ isOpen = true }: { isOpen: boolean }) => {
+
+    return (
+      <span>{isOpen ? CollapseOpenIcon() : CollapseClosedIcon()}</span>
+    )
+}
```
Comment (MrDSGC) [inline src/components/SignersPanel/CollapseButton.tsx]: missing line

File: src/components/Icons/CollapseClosedIcon.tsx [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+function CollapseClosedIcon() {
+    return (
+        <svg width="10" height="6" viewBox="0 0 10 6" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <path d="M1 1L5 5L9 1" stroke="#2F353F" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" />
+        </svg>
+    )
+}
+
+export default CollapseClosedIcon;
```
Comment (MrDSGC) [inline src/components/Icons/CollapseClosedIcon.tsx]: missing line

**Review by MrDSGC — COMMENTED** [review]
A few questions & comments

**Review by devbugging — COMMENTED** [review]
The presented functionality is on point. Nice work!

File: src/components/AccountAvatars.tsx (hunk 2 above)
Comment (bthaile) [reply src/components/AccountAvatars.tsx:59]: that is just how formatter did.

File: src/components/Arguments/index.tsx (hunk 3 above)
Comment (bthaile) [reply src/components/Arguments/index.tsx:124]: I didn't change any code, formatter changed it. Going to leave it

File: src/components/Icons/CollapseOpenIcon.tsx (hunk 4 above)
Comment (bthaile) [reply src/components/Icons/CollapseOpenIcon.tsx:3]: removed this component

File: src/components/Icons/CollapseOpenIcon.tsx (hunk 7 above)
Comment (bthaile) [reply src/components/Icons/CollapseOpenIcon.tsx:11]: not sure what "missing line" means. Are you saying missing "endline"? 

we need to decide about formatter. I'm using `yarn lint --fix` to determine formatting. We need to agree about this stuff.

File: src/components/Icons/CollapseClosedIcon.tsx (hunk 9 above)
Comment (bthaile) [reply src/components/Icons/CollapseClosedIcon.tsx]: see comment above.

**Review by MrDSGC — APPROVED** [review]

**Pull request merged**